### PR TITLE
Running inline tests with javascript/nodejs 

### PR DESCRIPTION
**Pull request by hhugo**
I'd like to run inline_tests (including expect tests) with JavaScript.

I was able to have something running with this PoC and https://github.com/ocsigen/js_of_ocaml/pull/817 
```
dune build @lib/tests/javascript-runtests
```

One of the constraint is that inline_tests that link with js_of_ocaml (the lib) can only be used in a JavaScript context. Ideally, I'd like to be able to specify that some tests run in JavaScript only, native only or both.

Do you have any suggestion on how to achieve that   ?

### CODE REVIEW

**Comment by diml** [comment]
Running inline tests in JavaScript seems good, thanks for looking at this!

Regarding how to specify which contexts to run the tests in, I suppose we could add a field `modes` to the `inline_tests` field. Then one could write `(library ... (inline_tests ... (modes js byte native)))`. The default would be `(modes best)`.

**Comment by hhugo** [comment]
Thanks. I've implemented your proposal. I'll try to write some tests later.

File: src/inline_tests.ml [hunk 1]
```diff
@@ -273,30 +320,46 @@ include Sub_system.Register_end_point(
...
+           in
+           Build.path exe >>>
+           Build.fanout
+             (Super_context.Deps.interpret sctx info.deps ~expander)
+             flags
+           >>^ fun (_deps, flags) ->
+           let exe, runner_args = match custom_runner with
+             | None -> exe, []
+             | Some runner ->
+               match Bin.which ~path:((Super_context.context sctx).path) runner with
```
Comment (diml) [inline src/inline_tests.ml]: We should use `Super_context.resolve_program` here

**Review by ghost — CHANGES_REQUESTED** [review]

**Comment by hhugo** [comment]
I've updated the PR with tests

**Comment by diml** [comment]
Thanks. I moved the tests to js_of_ocaml since they need nodejs to run (these are run as part of a separate `runtest-js` alias) and added some doc.

**Pull request merged**